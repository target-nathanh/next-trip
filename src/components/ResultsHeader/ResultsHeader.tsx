import React from 'react';
import { Container, makeStyles } from '@material-ui/core';
interface ResultsHeaderProps {
  stationName?: string;
  stopNumber?: number;
}
const useStyles = makeStyles((theme) => ({
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    backgroundColor: 'lightGray',
    padding: '20px',
    marginTop: '20px',
    marginBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      flexDirection: 'column',
    },
  },
  headerText: {
    fontWeight: 'bold',
  },
}));
const ResultsHeader: React.FC<ResultsHeaderProps> = (props: ResultsHeaderProps) => {
  const { header, headerText } = useStyles();
  return (
    <Container className={header}>
      <div className={headerText}>{props.stationName}</div>
      <div className={headerText}>Stop#: {props.stopNumber}</div>
    </Container>
  );
};

export default ResultsHeader;
