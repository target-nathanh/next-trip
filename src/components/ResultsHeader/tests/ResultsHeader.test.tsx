import React from 'react';
import { render } from '@testing-library/react';
import ResultsHeader from '../ResultsHeader';
import { MemoryRouter } from 'react-router-dom';

describe('ResultsHeader', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <ResultsHeader stationName={'Starblaster Station'} stopNumber={16137} />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
