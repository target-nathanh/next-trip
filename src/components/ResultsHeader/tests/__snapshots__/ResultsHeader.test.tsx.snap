// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsHeader matches snapshot 1`] = `
<div
  class="MuiContainer-root makeStyles-header-1 MuiContainer-maxWidthLg"
>
  <div
    class="makeStyles-headerText-2"
  >
    Starblaster Station
  </div>
  <div
    class="makeStyles-headerText-2"
  >
    Stop#: 
    16137
  </div>
</div>
`;
